### ISSUE TITLE
TFSAggregator not running on TFS 2015 update 1

### ISSUE THREAD

**Issue by IronGem** [tfsaggregator/tfsaggregator#139]
I have TFS 2015 update 1 and installed TFSAggregator2-2.1.173-Release.msi.  After building a policy and testing it, all was running smoothly until yesterday.  TFSAggregator is no longer making updates on Work Items that previously were triggering updates. I have been using Debug Viewer to monitor the Items that were triggering updates.  For most of yesterday and all of today there have been no trace logs caught in Debug View.  Nothing, that I can think of, has changed since it was working.  I have followed the Troubleshooting page. I have even updated TFSAggregator by uninstalling and installing the 2.1.178 msi.  The manual install verification Powershell Script displays 'Constructor Arguments - ["Release [TFS 2015.0]"}' even though I let the installer install the correct version for Update 1.  Debug View is still quiet and none of my Work Items are getting updated.  It sees as if TFS is not running the plugin at all.  
Is there anything I am missing? 
Please let me know if I can provide any further information.

**Comment by giuliov** [comment]
By reinstalling you forced a restart of the app, so it is not some kind of sticky memory state, but a configuration issue.
Sounds as if the installer senses the wrong TFS version, would be nice if you had the install log. If you can reinstall again by running this command line and attach the logs.
```
msiexec /x TFSAggregator-2.1.173-Release.msi /L*v uninstall.log
msiexec /i TFSAggregator-2.1.173-Release.msi /L*v install.log
```
also an export of registry content at `HKLM\SOFTWARE\Microsoft\TeamFoundationServer\14.0` can be useful.

TFS has breaking changes between RTM and Update 1, so in the Event Viewer Application log you should find some error when TFS tries to load the plugin and fails.

Sidenote, move to [release 2.1](https://github.com/tfsaggregator/tfsaggregator/releases/tag/v2.1.0).

**Comment by giuliov** [comment]
The log are _very_ verbose, avoid pasting the content in the issue, but use a Gist and refer them in the reply through a link.

**Comment by IronGem** [comment]
Thank you for the quick reply.  Your post prompted me to check the Application logs and I actually found the issue there.  
```
Detailed Message: The subscriber TFSAggregator2 raised an exception while being notified of event Microsoft.TeamFoundation.WorkItemTracking.Server.WorkItemChangedEvent.
Exception Message: Listener 'filelog' does not exist in the sharedListeners section. (type ConfigurationErrorsException)
```
I had added the Production Debugging code mentioned on the troubleshooting page into the web.config while troubleshooting a previous rule.  Apparently I didnt clean it up completely when I was finished and there were still remenents of the system.diagnostics node left.  That wasn't enough to prevent TFS from working, but apparently it did mess up the Aggregator's Notifier.  
After I finished removing the entire system.diagnostics node from the web.config, TFSAggregator started right back up without any complaints. 

Once again, thank you for the support and such a great product to simplify the lives of us poor sysadmins.  :)
